conan/cli/commands/compare.py

```diff
@@ -0,0 +1,233 @@
+import json
+import subprocess
+import os
+
+from conan.errors import ConanException
+from conan.api.conan_api import ConanAPI
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command
+from conan.internal.util.files import chdir
+from conan.api.model import RecipeReference, ListPattern
+
+def output_json(msg):
+    return json.dumps({"greet": msg})
+
+
+@conan_command(group="Security", formatters={"json": output_json})
+def compare(conan_api: ConanAPI, parser, *args):
+    """
+    Command to get the diff between versions
+    """
+
+    parser.add_argument("-op", "--old-path", help="Path to the old recipe")
+    parser.add_argument("-ov", "--old-reference", help='Old reference "mylib/1.0"')
+    parser.add_argument("-or", "--old-require", help='Old reference "mylib/1.0"')
+
+    parser.add_argument("-np", "--new-path", help="Path to the new recipe")
+    parser.add_argument("-nv", "--new-reference", help="New reference")
+    parser.add_argument("-nr", "--new-require", help='New reference "mylib/1.0"')
+
+    parser.add_argument("-s", "--split_diff", action="store_true")
+    parser.add_argument("--encoding", default="utf-8", help="Encoding to read diff")
+    parser.add_argument("-r", "--remote", action="append", default=None,
+                       help='Look in the specified remote or remotes server')
+
+    args = parser.parse_args(*args)
+
+    cwd = os.getcwd()
+    enabled_remotes = conan_api.remotes.list(args.remote or "*")
+
+    def _download_ref_from_remote(reference):
+        ref = RecipeReference().loads(reference)
+        full_ref, matching_remote = None, None
+        for remote in enabled_remotes:
+            if ref.revision:
+                no_rrev_ref = RecipeReference().loads(reference)
```

Comment:
```suggestion
                no_rrev_ref = RecipeReference.loads(reference)
```

That is a static method